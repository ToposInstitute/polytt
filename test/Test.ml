open Core

module S = Syntax

let header fname =
  String.make 20 '-' ^ "[" ^ fname ^ "]" ^ String.make 20 '-' ^ "\n"

let rec execute_file cwd fname =
  let abs_fname = Filename.concat cwd fname in
  if String.equal (Filename.extension abs_fname) ".poly" then
    try
      let _ = print_string (header fname) in
      ignore @@ Loader.load abs_fname false;
    with
      e ->
      Format.eprintf "Could not load file %s@." fname;
      raise e
  else if Sys.is_directory abs_fname then
    Array.iter (execute_file abs_fname) (Sys.readdir abs_fname)

let () =
  let polytt_files = Sys.readdir "." in
  Array.sort String.compare polytt_files;
  Array.iter (execute_file (Sys.getcwd ())) polytt_files
