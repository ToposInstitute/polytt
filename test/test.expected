--------------------[basics.poly]--------------------

Normal Form: 5
Normal Form: ℕ × ℕ
Normal Form: ℕ × ℕ
(λ P → (_ : base P) × fib P _) : Poly → Poly
(λ P → λ ~> FIXME λ a⁺ → (a⁺ , λ <2> → <2>) FIXME) : (P : Poly) → (_ : base P) × fib P _ ⇒ (_ : base P) × fib P _
Normal Form: (42 , λ <3> → succ (succ <3>))
Normal Form: (12 , λ <2> → succ (succ <2>))
Normal Form: (12 , λ <8> → succ (succ (succ (succ <8>))))
Normal Form: λ x _ → x
Normal Form: λ ~> FIXME λ n⁺ → (n⁺ , λ <1> → succ n⁺) FIXME
Normal Form: ((1 , 12) , λ <2> → succ (succ (succ (snd <2>))))
Normal Form: ((2 , 1) , λ <3> → (snd <3> , fst <3>))
Normal Form: λ P Q → λ ~> FIXME λ pf⁺ → (fst (snd pf⁺ (fst pf⁺)) , λ <7> → (snd (snd pf⁺ (fst pf⁺)) <7> , (fst pf⁺ , <7>))) FIXME
Normal Form: (12 , λ <7> → (succ (succ <7>) , (3 , <7>)))
Normal Form: λ P Q → λ ~> FIXME λ pq⁺ → ((fst pq⁺ , λ _ → snd pq⁺) , λ <3> → (fst <3> , snd <3>)) FIXME
Normal Form: λ S T A B → #{ .get, .set } → { .get = S → A, .set = S → B → T } _
--------------------[test-let.poly]--------------------

Normal Form: 9
--------------------[a.poly]--------------------

--------------------[b.poly]--------------------

--------------------[c.poly]--------------------

baz : ℕ
